pkg/conversion/scheme.go

```diff
@@ -19,22 +19,19 @@ package conversion
 import (
 	"fmt"
 	"reflect"
+
+	"k8s.io/kubernetes/pkg/api/unversioned"
 )
 
 // Scheme defines an entire encoding and decoding scheme.
 type Scheme struct {
 	// versionMap allows one to figure out the go type of an object with
 	// the given version and name.
-	versionMap map[string]map[string]reflect.Type
-
-	// typeToVersion allows one to figure out the version for a given go object.
-	// The reflect.Type we index by should *not* be a pointer. If the same type
-	// is registered for multiple versions, the last one wins.
-	typeToVersion map[reflect.Type]string
+	gvkToType map[unversioned.GroupVersionKind]reflect.Type
 
-	// typeToKind allows one to figure out the desired "kind" field for a given
-	// go object. Requirements and caveats are the same as typeToVersion.
-	typeToKind map[reflect.Type][]string
+	// typeToGroupVersion allows one to find metadata for a given go object.
+	// The reflect.Type we index by should *not* be a pointer.
+	typeToGVK map[reflect.Type][]unversioned.GroupVersionKind
```

Comment:
same general comment here on acronyms.

i think the api server code is hard enough to follow that abbreviating GroupVersionKind makes it worse for new eyes.

also would need to align the name with teh comment.